src/modules/form/hooks/rhf/useEnrichedFormData.tsx

```diff
@@ -60,9 +68,18 @@ export const useEnrichedFormData = <T extends FieldValues>({
   });
 
   // supplement with data from picklist
-  const [result, setResult] = useState<DefaultValues<T> | undefined>(undefined);
+  const [enrichedDefaultValues, setEnrichedDefaultValues] = useState<
+    DefaultValues<T> | undefined
+  >(undefined);
+
   useEffect(() => {
-    if (loading || result) return;
+    // Form has no remote picklists; nothing to enrich
+    if (!hasRemotePicklists) return;
+    // Remote picklists are loading, wait
+    if (loading) return;
+    // Default values have already been enriched
+    if (!!enrichedDefaultValues) return;
+
     const mutation = cloneDeep(defaultValues);
```

Comment:
Not changed by your PR or related to this bugfix, so feel free to ignore, but, I found this var name `mutation` a bit confusing. `mutation` immediately makes me think graphQL, so it's just a bit of extra cognitive load to scan through the code and understand, oh right, there is no actual backend data mutation happening.